### PR TITLE
snap-confine: build with `-Wmissing-field-initializers`

### PR DESCRIPTION
**Pull request by mvo5**
The tiobe software quality checker flaged this and while it's a bit debatable it adds some clarity around the booleans in the `sc_mount` struct so I think it's overall worth it.

### CODE REVIEW

**Review by alexmurray — APPROVED** [review]
I think this is a worthy addition - it keeps the code more readable too.

**Review by ZeyadYasser — APPROVED** [review]
This increases readability imo, Thank you!

File: cmd/Makefile.am [hunk 1]
```diff
@@ -8,7 +8,8 @@ noinst_PROGRAMS =
 noinst_LIBRARIES =
 
 CHECK_CFLAGS = -Wall -Wextra -Wmissing-prototypes -Wstrict-prototypes \
-	-Wno-missing-field-initializers -Wno-unused-parameter
+	-Wno-missing-field-initializers -Wno-unused-parameter \
```
Comment (valentindavid) [inline cmd/Makefile.am:11]: Can you remove `-Wno-missing-field-initializers`?

Comment (mvo5) [reply cmd/Makefile.am]: Indeed, thank you, nice catch! Indeed `-Wmissing-field-initializers`  is already included in `-Wextra` so I removed both `-W{no-,}missing-field-initializers`.

**Review by valentindavid — APPROVED** [review]

**Pull request merged**